.github/workflows/codeql.yml

```diff
@@ -0,0 +1,43 @@
+name: "Code Scanning - Action"
+
+on:
+  pull_request:
```

Comment:
Let's use the [same configuration](https://github.com/letsencrypt/boulder/blob/8e156f4dd96b3aaeb4e9250d76ed34a0730ac4bd/.github/workflows/boulder-ci.yml#L5-L14) as our main `boulder-ci` job: Run this on all pull requests, and also on pushes to `main` and to `release-branch-*`.